python/ray/rllib/agents/dqn/dqn.py

```diff
@@ -160,6 +167,12 @@ def _init(self):
             if k not in self.config["optimizer"]:
                 self.config["optimizer"][k] = self.config[k]
 
+        if self.config.get("parameter_noise", False):
+            self.config["callbacks"]["on_episode_start"] = tune.function(
+                on_episode_start)
```

Comment:
This will override the callbacks the user sets right? Can we make sure to call the original ones afterwards?

Reply:
yes we must call them. I found that the callback functions are not allowed to include sth like self.config["callbacks"]["on_episode_start"]. They will cause pickle problem. I validated my implementation with some simple user specified callbacks.

Reply:
Sounds good. I removed the .func from the start one if that's ok. It wasn't needed for the end one?

Reply:
yes, actually no need for `.func`.